terraform/modules/r53-dns-firewall/main.tf

```diff
@@ -0,0 +1,186 @@
+# R53 Resolver DNS Firewall Resources - created on a per-VPC basis
+
+# Rule group
+resource "aws_route53_resolver_firewall_rule_group" "this" {
+  name = "${var.tags_prefix}-r53-dns-firewall-rule-group"
+  tags = var.tags_common
+}
+
+# Allowed domain list
+resource "aws_route53_resolver_firewall_domain_list" "allow" {
+  name    = "${var.tags_prefix}-allow-domain-list"
+  domains = [for domain in var.allowed_domains : "${domain}."]
+  tags    = var.tags_common
+}
+
+# Blocked domain list
+resource "aws_route53_resolver_firewall_domain_list" "block" {
+  name    = "${var.tags_prefix}-block-domain-list"
+  domains = [for domain in var.blocked_domains : "${domain}."]
+  tags    = var.tags_common
+}
+
+# Default rule to ALERT on AWS-managed bad domain lists - see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resolver-dns-firewall-managed-domain-lists.html
+# These rules will be set to ALERT initially and enabled in production - after a period of monitoring we will switch to BLOCK
+data "external" "aws_managed_domain_lists" {
+  program = ["bash", "${path.module}/fetch-aws-managed-domain-lists.sh"]
+}
+resource "aws_route53_resolver_firewall_rule" "default_alert" {
+  for_each = data.external.aws_managed_domain_lists.result
+
+
+  action                  = "ALERT"
+  firewall_domain_list_id = each.value
+  priority                = each.key == "AWSManagedDomainsAggregateThreatList" ? 2 : each.key == "AWSManagedDomainsMalwareDomainList" ? 3 : each.key == "AWSManagedDomainsBotnetCommandandControl" ? 4 : 5
+  firewall_rule_group_id  = aws_route53_resolver_firewall_rule_group.this.id
+  name                    = "Alert_${each.key}"
+  # block_response          = var.block_response # Leaving this commented out until we turn on blocking in a future PR
```

Comment:
Personally I'd remove this as it's non-functional and would conflict with `action = "ALERT'`.